core/services/functions/external_adapter_client_test.go

```diff
@@ -0,0 +1,244 @@
+package functions
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/smartcontractkit/chainlink/v2/core/internal/testutils"
+)
+
+func Test_FetchEncryptedSecrets_Success(t *testing.T) {
```

Comment:
Can you convert every test into a single helper function call along the lines of "runTest(adapterJSONResponse, expectedSecrets, expectedUserError, expectedError)"?